File: BackEnd/Api/Domain/IService/IStorageAreaService.cs
using Api.Application.DataTransfer;
using Api.Application.DataTransfer.Filters;
using Api.Infrastructure.Utilities;

namespace Api.Application.Services;

public interface IStorageAreaService
{
	Task<IEnumerable<StorageAreaDto>> GetStorageAreas();
	Task<StorageAreaDto> GetStorageAreaByCode(string code);
	Task<StorageAreaDto> CreateStorageArea(CreateStorageAreaDto createStorageAreaDto);
	Task<StorageAreaDto> UpdateStorageArea(string id, CreateStorageAreaDto updateStorageAreaDto);
	Task<Page<StorageAreaDto>> FilterStorageAreas(StorageAreaFilter filter);
	Task<int> CountStorageAreasAsync();
}

File: FrontEnd/src/service/IService/IStorageAreaService.ts
import type { StorageAreaDto } from "@/model/dto/StorageAreaDto";
import type { Filter, Page } from "@/model/Page";
import type { StorageArea } from "@/model/StorageArea";

export interface IStorageAreaService {
	getStorageAreas(filtering?: Filter<{ nameCode: string }>): Promise<Page<StorageArea>>;
	getStorageAreaById(id: string): Promise<StorageArea | undefined>;
	
	createStorageArea(storageArea: StorageArea): Promise<StorageArea>;
	updateStorageArea(storageArea: StorageArea): Promise<StorageArea>;
	
	getNumberOfStorageAreas(): Promise<number>;
}